### PR TITLE
Add `setLineString` and `getLineString` methods

### PR DESCRIPTION
**Pull request by dtrucs**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -915,6 +924,165 @@ export default class MapboxPathControl implements IControl {
...
+    const { features } = this.getFeatureCollection();
+    const { coordinates, points, paths } = features
+      .sort((a, b) => {
+        if (
+          a.geometry.type === "LineString" &&
+          a.properties!.index === b.properties!.index &&
+          b.properties!.isDeparture === true
+        ) {
+          return 1;
+        }
```
Comment (juliensnz) [inline src/index.ts:1057]:
```suggestion
        const isSameDepartureLine = a.geometry.type === "LineString" &&
          a.properties!.index === b.properties!.index &&
          b.properties!.isDeparture === true;
        if (isSameDeparturLine) {
          return 1;
        }
```

File: src/index.ts [hunk 2]
```diff
@@ -915,6 +924,165 @@ export default class MapboxPathControl implements IControl {
...
+                ]
+              );
+              lineStringify.paths.push("junction");
+            } else if (feature.properties!.isFollowingDirections) {
+              lineStringify.paths.push("direction");
+            } else {
+              lineStringify.paths.push("free");
+            }
+          }
+          return lineStringify;
```
Comment (juliensnz) [inline src/index.ts]:
```suggestion
          switch (feature.geometry.type) {
            case 'Point':
              return addGeometry(lineStringify, feature);
            case 'LineString':
              lineStringify = addFirstCoordinate(lineStringify, feature)

              if (feature.properties!.isPhantomJunction) {
                lineStringify = addDeparture(lineStringify, feature);
                lineStringify = addJunction(lineStringify);
              } else if (feature.properties!.isFollowingDirections) {
                lineStringify = addDirection(lineStringify);
              } else {
                lineStringify = addFree(lineStringify);
              }

              return lineStringify;
            default:
              return lineStringify
          }
```

File: dist/index.js [hunk 3]
```diff
@@ -7086,6 +7089,115 @@ class MapboxPathControl {
             ],
         };
     }
+    setLineString(feature) {
+        var _a, _b;
```
Comment (jrmi) [inline dist/index.js:7093]: If think that even in typescript, `let` should be prefered as `var` relative to is scope ?

File: dist/index.js [hunk 4]
```diff
@@ -7086,6 +7089,115 @@ class MapboxPathControl {
...
+        const coordinates = [...feature.geometry.coordinates];
+        const referenceToBuildFeatureLineString = (_b = (_a = feature.properties.points) === null || _a === void 0 ? void 0 : _a.flatMap((item, index) => [item, feature.properties.paths[index]].filter(Boolean))) !== null && _b !== void 0 ? _b : [
```
Comment (jrmi) [inline dist/index.js:7095]: Can you split this instruction for readability purpose ?

File: dist/index.js [hunk 5]
```diff
@@ -7086,6 +7089,115 @@ class MapboxPathControl {
...
+            coordinates[0],
+            this.isFollowingDirections ? "direction" : "free",
+            coordinates[coordinates.length - 1],
+        ];
+        const points = referenceToBuildFeatureLineString
+            .filter((nextCoordinates, index, array) => Array.isArray(nextCoordinates) &&
+            !((array[index - 1] === "junction" &&
```
Comment (jrmi) [inline dist/index.js:7102]: Can you add a comment to tell what you try to filter here ?

File: dist/index.js [hunk 6]
```diff
@@ -7086,6 +7089,115 @@ class MapboxPathControl {
...
+            type: "Feature",
+            geometry: {
+                coordinates: nextCoordinates,
+                type: "Point",
+            },
+            properties: {
+                index,
+            },
+        }));
+        const lines = referenceToBuildFeatureLineString.reduce((memo, item, index, array) => {
```
Comment (jrmi) [inline dist/index.js:7116]: I think that a so big reduce should have a comments to explain it ?

**Review by jrmi — APPROVED** [review]

**Pull request merged**